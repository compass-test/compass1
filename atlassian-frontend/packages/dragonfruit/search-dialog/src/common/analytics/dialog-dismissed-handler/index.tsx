export { DialogDismissedHandler } from './dialog-dismissed-handler';
