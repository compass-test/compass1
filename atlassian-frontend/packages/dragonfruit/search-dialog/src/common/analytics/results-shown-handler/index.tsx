export { default as SearchResultsShownHandler } from './results-shown-handler';
