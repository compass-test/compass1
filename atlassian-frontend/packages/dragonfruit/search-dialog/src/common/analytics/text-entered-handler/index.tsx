export { default as TextEnteredHandler } from './text-entered-handler';
