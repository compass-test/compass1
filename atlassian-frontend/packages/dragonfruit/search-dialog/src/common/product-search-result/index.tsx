export { ProductSearchResultForPreQuery } from './product-search-result';
export { default as ProductSearchResult } from './product-search-result';
