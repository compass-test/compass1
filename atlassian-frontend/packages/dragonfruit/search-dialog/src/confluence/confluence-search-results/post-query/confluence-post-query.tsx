import React from 'react';
import {
  LinkComponent,
  SearchResultSection,
  SearchResultSectionLink,
} from '@atlassian/search-dialog';
import { FormattedMessage, InjectedIntlProps, injectIntl } from 'react-intl';
import {
  ConfItemResults,
  ConfPeopleResults,
  ConfSpaceResults,
  withClients,
} from '../../clients';
import { ConfItemResult } from '../../clients/response-types';
import { messages } from '../../../messages';
import {
  SearchResultsShownHandler,
  withAnalytics,
} from '../../../common/analytics';
import {
  AdvancedSearchAnalyticsProps,
  onAdvancedSearchSelected,
  onShowAllClicked,
  onShowMoreClicked,
  OnShowMoreClickedProps,
  ScreenType,
} from '../../../common/analytics/events';
import { ResultAvatar as Avatar } from '../../../common/result-avatar';
import { addQuery } from '../../../utils/url-utils';
import { ProductSearchResult } from '../../../common/product-search-result';
import { getResultsDetailsAnalyticData } from '../confluence-search-results-utils';
import { usePrimarySiteConfluenceAdvancedSearchUrlFactory } from '../../confluence-utils/confluence-url-utils';
import { useSearchSessionId } from '../../../common/search-session-provider';
import {
  CompassComponentType,
  CompassSearchComponentNodeFragment,
} from '@atlassian/dragonfruit-graphql';
import { ComponentTypeIcon } from '@atlassian/dragonfruit-components';
import { TeamDetails } from '@atlassian/dragonfruit-rest';
import { User } from '@atlaskit/user-picker';
import {
  getTeamOrUserAvatar,
  getTeamOrUserDisplayName,
  getTeamOrUserURL,
} from '../utils';

interface Props {
  services: {
    isLoading: boolean;
    cursor: string | null;
    results: CompassSearchComponentNodeFragment[] | null;
  };
  libsAppsMore: {
    isLoading: boolean;
    cursor: string | null;
    results: CompassSearchComponentNodeFragment[] | null;
  };
  peopleTeams: {
    isLoading: boolean;
    results: (TeamDetails | User)[] | null;
  };
  items: {
    isLoading: boolean;
    results: ConfItemResults | null;
  };
  spaces: {
    isLoading: boolean;
    results: ConfSpaceResults | null;
  };
  people: {
    isLoading: boolean;
    results: ConfPeopleResults | null;
  };
  query: string;
  advancedSearchSelected: (props: AdvancedSearchAnalyticsProps) => any;
  showMoreClicked: (props: OnShowMoreClickedProps) => any;
  showAllClicked: (props: OnShowMoreClickedProps) => any;
  linkComponent?: LinkComponent;
  formatDate?: (lastModified: string) => JSX.Element;
  isLoading: boolean;
  screenType: ScreenType;
  isCollapsed: boolean;
  siteUrl?: string;
  fetchMoreFn: (typeResult: string, cursor: string | null) => void;
  teamNames: { [key: string]: TeamDetails | null };
}

interface ContextProps {
  advancedSearchUrl: string;
  searchSessionId: string;
}

enum TypeResults {
  SERVICES = 'SERVICES',
  LIBS_APPS_MORE = 'LIBS_APPS_MORE',
}

interface State {
  query: string;
  isShowingMore: boolean;
  serviceShowIndex: number;
  libsAppsMoreShowIndex: number;
  typeResult: string;
}

// For tests only! Don't use outside this file.
export const POST_QUERY_SPACE_LIMIT = 3;
export const POST_QUERY_PEOPLE_LIMIT = 3;

// const getItemResults = (
//   isShowingMore: boolean,
//   items: ConfItemResults | null,
// ) => {
//   if (!items) {
//     return null;
//   }

//   const itemResults = isShowingMore
//     ? (items && items.items) || []
//     : (items && items.items.slice(0, INITIAL_POST_QUERY_ITEM_LIMIT)) || [];

//   return {
//     ...items,
//     items: itemResults,
//   };
// };

class ConfluencePostQueryBase extends React.Component<
  Props & InjectedIntlProps & ContextProps,
  State
> {
  state = {
    isShowingMore: false,
    query: '',
    serviceShowIndex: 0,
    libsAppsMoreShowIndex: 0,
    typeResult: TypeResults.SERVICES,
  };

  static getDerivedStateFromProps(props: Props, state: State) {
    if (props.query !== state.query) {
      return {
        ...state,
        isShowingMore: false,
        query: props.query,
      };
    }
    return null;
  }

  getTimeDetailForConfluenceObjectResult = (result: ConfItemResult) => {
    const { formatDate, intl } = this.props;

    if (result.isRecentResult) {
      const message = intl.formatMessage(
        messages.common_recently_viewed_section_heading,
      );
      return <span title={message}>{message}</span>;
    }

    if (formatDate && result.lastModified) {
      return formatDate(result.lastModified);
    }

    return undefined;
  };
  // NOTE: may need these for the advancedSearchPage
  // onShowMore = () => {
  //   const { showMoreClicked, items } = this.props;

  //   const itemResults = getItemResults(false, items.results);

  //   showMoreClicked({
  //     total: itemResults?.totalSize ?? 0,
  //     currentSize: itemResults?.items.length ?? 0,
  //     pageSize: INITIAL_POST_QUERY_ITEM_LIMIT,
  //   });

  //   this.setState({
  //     isShowingMore: true,
  //   });
  // };

  onShowMoreServices = () => {
    this.props.fetchMoreFn(TypeResults.SERVICES, this.props.services.cursor);
    // this.setState({
    //   serviceShowIndex: this.state.serviceShowIndex + 1,
    // });
  };

  onShowMoreLibsAppsMore = () => {
    this.props.fetchMoreFn(
      TypeResults.LIBS_APPS_MORE,
      this.props.libsAppsMore.cursor,
    );
    // this.setState({
    //   libsAppsMoreShowIndex: this.state.libsAppsMoreShowIndex + 1,
    //   type: TypeResults.LIBS_APPS_MORE
    // });
  };

  // NOTE: may need these for the advancedSearchPage
  // onShowAll = () => {
  //   const { showAllClicked, items } = this.props;

  //   const itemResults = getItemResults(true, items.results);

  //   showAllClicked({
  //     total: itemResults?.totalSize ?? 0,
  //     currentSize: itemResults?.items.length ?? 0,
  //     pageSize: INITIAL_POST_QUERY_ITEM_LIMIT,
  //   });
  // };

  getShowMoreLink = (
    // index: number,
    // limit: number,
    // totalItemCount: number,
    // advancedSearchUrl: string,
    fn: () => void,
    cursor: string | null,
  ) => {
    if (!cursor) {
      return null;
    } else {
      return (
        <SearchResultSectionLink onClick={fn} id="show-more-button">
          <FormattedMessage {...messages.common_show_more_results_link} />
        </SearchResultSectionLink>
      );
    }
    // NOTE: may need these for the advancedSearchPage
    // TODO if hasNextPage and endCursor -> showmore
    // console.log((index + 1) * INITIAL_POST_QUERY_ITEM_LIMIT, totalItemCount, limit)
    // cases when we don't show a show more button
    // if ((index + 1) * INITIAL_POST_QUERY_ITEM_LIMIT >= totalItemCount) {
    //   return null;
    // }

    // if ((index + 1) * INITIAL_POST_QUERY_ITEM_LIMIT > limit) {
    //   return (
    //     <ConfluenceSearchResultSectionLink
    //       onClick={this.onShowAll}
    //       linkComponent={this.props.linkComponent}
    //       href={advancedSearchUrl}
    //       id="show-all-link"
    //     >
    //       <FormattedMessage {...messages.common_show_all_results_link} />
    //     </ConfluenceSearchResultSectionLink>
    //   );
    // } else {
    //   return (
    //     <SearchResultSectionLink onClick={fn} id="show-more-button">
    //       <FormattedMessage {...messages.common_show_more_results_link} />
    //     </SearchResultSectionLink>
    //   );
    // }
  };

  getProductSearchResults = (
    components: CompassSearchComponentNodeFragment[],
    searchSessionId: string,
    screenType: ScreenType,
    isCollapsed: boolean,
    sectionIndex: number,
    totalNumberOfComponentsInPreviousSections: number,
    linkComponent?: LinkComponent,
  ) => {
    const { teamNames } = this.props;
    return components.map((result, index) => {
      const ownerId = result.component?.ownerId;
      return (
        <ProductSearchResult
          key={`item_${result.link}`}
          title={result.component?.name ?? 'error'}
          href={addQuery(result.link, 'search_id', searchSessionId)}
          icon={
            <ComponentTypeIcon
              label={result.component?.name}
              type={result.component?.type || CompassComponentType.OTHER}
            />
          }
          containerDetail={result.component?.description || undefined}
          owner={
            teamNames && ownerId && teamNames[ownerId]
              ? teamNames[ownerId]!.displayName
              : undefined
          }
          timeDetail={undefined}
          linkComponent={linkComponent}
          screen={screenType}
          isCollapsed={isCollapsed}
          analyticContext={{
            id: result.component?.id,
            sectionIndex,
            sectionId: `compass-${
              result.component?.type.toLowerCase() ||
              CompassComponentType.OTHER.toLowerCase()
            }` as any,
            containerId: 'none',
            globalIndex: totalNumberOfComponentsInPreviousSections + index,
            indexWithinSection: index,
            isCached: false,
          }}
        />
      );
    });
  };

  renderResultsSection = ({
    results,
    totalResultsCount,
    sectionIndex,
    totalNumberOfResultsInPreviousSections,
    message,
    showMoreIndex,
    showMoreFunc,
    cursor,
  }: {
    results: CompassSearchComponentNodeFragment[];
    totalResultsCount: number;
    sectionIndex: number;
    // This is the total number of items that were rendered in sections that appear before this
    totalNumberOfResultsInPreviousSections: number;
    message: { id: string; defaultMessage: string; description: string };
    showMoreIndex: number;
    showMoreFunc: () => void;
    cursor: string | null;
  }) => {
    const {
      // advancedSearchUrl,
      linkComponent,
      searchSessionId,
      screenType,
      isCollapsed,
    } = this.props;

    return results.length ? (
      <SearchResultSection
        // totalResults={totalResultsCount}
        title={<FormattedMessage {...message} />}
      >
        {this.getProductSearchResults(
          results,
          searchSessionId,
          screenType,
          isCollapsed,
          sectionIndex,
          totalNumberOfResultsInPreviousSections,
          linkComponent,
        )}
        {this.getShowMoreLink(
          // showMoreIndex,
          // 100,
          // totalResultsCount,
          // advancedSearchUrl,
          showMoreFunc,
          cursor,
        )}
      </SearchResultSection>
    ) : null;
  };

  renderPeopleSection = ({
    peopleResults,
    sectionIndex,
    totalNumberOfItemsInPreviousSections,
  }: {
    peopleResults: (TeamDetails | User)[];
    sectionIndex: number;
    // This is the total number of items that were rendered in sections that appear before this
    totalNumberOfItemsInPreviousSections: number;
  }) => {
    const {
      linkComponent,
      searchSessionId,
      screenType,
      isCollapsed,
      isLoading,
    } = this.props;

    if (isLoading) {
      return null;
    }

    return peopleResults.length ? (
      <SearchResultSection
        title={
          <FormattedMessage
            {...messages.compass_teams_and_people_section_heading}
          />
        }
      >
        {peopleResults.map((result, index) => (
          <ProductSearchResult
            key={`people_${result.id}`}
            title={getTeamOrUserDisplayName(result)}
            href={addQuery(
              getTeamOrUserURL(result),
              'search_id',
              searchSessionId,
            )}
            icon={
              <Avatar
                borderColor="transparent"
                src={getTeamOrUserAvatar(result)}
                appearance="circle"
                size={isCollapsed ? 'small' : 'xsmall'}
              />
            }
            linkComponent={linkComponent}
            screen={screenType}
            isCollapsed={isCollapsed}
            analyticContext={{
              id: result.id,
              sectionIndex,
              sectionId: `compass-${
                (result as any).type ? 'user' : 'team'
              }` as any,
              containerId: null,
              globalIndex: totalNumberOfItemsInPreviousSections + index,
              indexWithinSection: index,
              isCached: false,
            }}
          />
        ))}
      </SearchResultSection>
    ) : null;
  };

  render() {
    const { services, libsAppsMore, peopleTeams, isLoading } = this.props;

    const serviceResults = services.results ?? [];
    const totalServiceCount = services.results?.length || 0;

    let currentSectionIndex = 0;
    let totalNumberOfItemsInPreviousSections = 0;

    const serviceSection = this.renderResultsSection({
      results: serviceResults,
      totalResultsCount: totalServiceCount,
      sectionIndex: currentSectionIndex,
      totalNumberOfResultsInPreviousSections: totalNumberOfItemsInPreviousSections,
      message: messages.compass_services_section_heading,
      showMoreIndex: this.state.serviceShowIndex,
      showMoreFunc: this.onShowMoreServices,
      cursor: services.cursor,
    });

    currentSectionIndex = serviceSection
      ? currentSectionIndex + 1
      : currentSectionIndex;
    totalNumberOfItemsInPreviousSections += totalServiceCount;

    const libsAppsMoreResults = libsAppsMore.results ?? [];
    const totalLibsAppsMoreCount = libsAppsMore.results?.length || 0;

    const libsAppsMoreSection = this.renderResultsSection({
      results: libsAppsMoreResults,
      totalResultsCount: totalLibsAppsMoreCount,
      sectionIndex: currentSectionIndex,
      totalNumberOfResultsInPreviousSections: totalNumberOfItemsInPreviousSections,
      message: messages.compass_libraries_applications_more_section_heading,
      showMoreIndex: this.state.libsAppsMoreShowIndex,
      showMoreFunc: this.onShowMoreLibsAppsMore,
      cursor: libsAppsMore.cursor,
    });

    const peopleTeamsSection = this.renderPeopleSection({
      peopleResults: peopleTeams.results || [],
      sectionIndex: currentSectionIndex,
      totalNumberOfItemsInPreviousSections,
    });

    return (
      <span data-test-id="search-dialog-compass-post-query">
        <div
          data-test-id={'search-result-section-services'}
          className={'search-result-section'}
        >
          {serviceSection}
        </div>
        <div
          data-test-id={'search-result-section-appsnlibs'}
          className={'search-result-section'}
        >
          {libsAppsMoreSection}
        </div>
        <div
          data-test-id={'search-result-section-teams'}
          className={'search-result-section'}
        >
          {peopleTeamsSection}
        </div>
        <SearchResultsShownHandler
          isPreQuery={false}
          isLoading={isLoading}
          resultCount={totalNumberOfItemsInPreviousSections}
          sectionCount={currentSectionIndex}
          results={getResultsDetailsAnalyticData([], [], false)}
        />
      </span>
    );
  }
}

export const ConfluencePostQuery = injectIntl(
  (props: Props & InjectedIntlProps) => {
    const searchSessionId = useSearchSessionId();
    const confluenceUrlFactory = usePrimarySiteConfluenceAdvancedSearchUrlFactory();

    return (
      <ConfluencePostQueryBase
        {...props}
        searchSessionId={searchSessionId}
        advancedSearchUrl={confluenceUrlFactory(props.query)}
      />
    );
  },
);

export default withAnalytics({
  advancedSearchSelected: onAdvancedSearchSelected,
  showMoreClicked: onShowMoreClicked,
  showAllClicked: onShowAllClicked,
})(withClients(ConfluencePostQuery));
