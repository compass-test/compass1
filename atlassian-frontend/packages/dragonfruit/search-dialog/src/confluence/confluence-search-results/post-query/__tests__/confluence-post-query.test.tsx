import React, { ReactElement } from 'react';
import {
  assertSearchResultSectionsHaveTitles,
  getSearchResultSectionTitleText,
  shallowWithIntl,
} from '../../../../__tests__/__fixtures__/intl-test-helpers';
import {
  createPeopleTeamsResult,
  createServiceResults,
} from '../../../../__tests__/__fixtures__/mock-search-results';
import { ConfluencePostQuery } from '../confluence-post-query';
import {
  SearchResultSection,
  SearchResultSectionLink,
} from '@atlassian/search-dialog';
import { messages } from '../../../../messages';
import { SearchResultsShownHandler } from '../../../../common/analytics';
import { ScreenType } from '../../../../common/analytics/events';
import { ProductSearchResult } from '../../../../common/product-search-result';
import { CompassComponentType } from '@atlassian/dragonfruit-graphql';

jest.mock('@atlassian/search-dialog', () => ({
  SearchResultSection: (props: any) => <div {...props} />,
  SearchResultSectionLink: () => <div />,
  Link: () => 'a',
}));

jest.mock('../../../../common/product-search-result', () => ({
  ProductSearchResult: () => <div />,
}));

jest.mock('../../../../common/search-session-provider', () => ({
  useSearchSessionId: () => 'abc',
}));

jest.mock('../../../confluence-features', () => ({
  useFeatures: () => ({ isMultiSite: false }),
}));

jest.mock('../../../confluence-utils/confluence-url-utils', () => ({
  usePrimarySiteConfluenceAdvancedSearchUrlFactory: () => () =>
    'some/advanced/search/url',
}));

describe('<ConfluencePostQuery />', () => {
  const names = {
    '1eb19c9d-e4c7-45c8-aec3-8a2d7cbe5515': {
      displayName: 'this is a team',
    } as any,
    '15c3f236-8339-43e0-9049-2fd03e808d9d': {
      displayName: 'another team',
    } as any,
    '5a740f67-ee17-4558-b567-60116bcd266d': {
      displayName: 'yet another team',
    } as any,
  };
  const createResults = (num: number = 1) => ({
    items: {
      isLoading: false,
      results: null,
    },
    spaces: {
      isLoading: false,
      results: null,
    },
    people: {
      isLoading: false,
      results: null,
    },
    services: {
      isLoading: false,
      cursor: null,
      results: createServiceResults(num, CompassComponentType.SERVICE),
      teamNames: names,
    },
    libsAppsMore: {
      isLoading: false,
      cursor: null,
      results: createServiceResults(num, CompassComponentType.OTHER),
      teamNames: names,
    },
    peopleTeams: {
      isLoading: false,
      results: createPeopleTeamsResult(num),
    },
  });

  const advancedSearchSelected = jest.fn();
  const showMoreClicked = jest.fn();
  const showAllClicked = jest.fn();
  const fetchMoreFn = jest.fn();

  const baseProps = {
    query: 'abcdef!@#$%^',
    advancedSearchUrl: 'advanced/search/link',
    advancedSearchSelected,
    showMoreClicked,
    showAllClicked,
    searchSessionId: 'testSearchSessionId',
    formatDate: (date: string) => <>{date}</>,
    screenType: 'postQuerySearchResults' as ScreenType,
    isLoading: false,
    isCollapsed: false,
    enabledFilters: [],
    items: {
      isLoading: false,
      results: {
        items: [],
        timings: 0,
        totalSize: 0,
      },
    },
    spaces: {
      isLoading: false,
      results: {
        items: [],
        timings: 0,
      },
    },
    people: {
      isLoading: false,
      results: {
        items: [],
        timings: 0,
      },
    },
    services: {
      isLoading: false,
      cursor: null,
      results: [],
      teamNames: {},
    },
    libsAppsMore: {
      isLoading: false,
      cursor: null,
      results: [],
      teamNames: {},
    },
    peopleTeams: {
      isLoading: false,
      results: [],
    },
    isMultiSite: false,
  };

  const shallow = <P extends {}>(
    node: ReactElement<P>,
    additionalOptions: {} = {},
  ) => {
    return shallowWithIntl(node, additionalOptions).dive();
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('render with results matches snapshot - with team names', () => {
    const wrapper = shallow(
      <ConfluencePostQuery
        {...baseProps}
        query="query"
        teamNames={names}
        fetchMoreFn={fetchMoreFn}
        {...createResults()}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('render with results matches snapshot', () => {
    const wrapper = shallow(
      <ConfluencePostQuery
        {...baseProps}
        query="query"
        teamNames={{}}
        fetchMoreFn={fetchMoreFn}
        {...createResults()}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  describe('with 20 results', () => {
    it('renders item results properly', () => {
      const wrapper = shallow(
        <ConfluencePostQuery
          {...baseProps}
          services={{
            isLoading: false,
            cursor: null,
            results: createServiceResults(31, CompassComponentType.SERVICE),
          }}
          teamNames={{}}
          fetchMoreFn={fetchMoreFn}
        />,
      );

      expect(wrapper.find(ProductSearchResult)).toHaveLength(31);
    });

    it('fires search results shown with 10 results and 1 section', () => {
      const wrapper = shallow(
        <ConfluencePostQuery
          {...baseProps}
          services={{
            isLoading: false,
            cursor: null,
            results: createServiceResults(10, CompassComponentType.SERVICE),
          }}
          teamNames={{}}
          fetchMoreFn={fetchMoreFn}
        />,
      );

      expect(wrapper.find(SearchResultsShownHandler).prop('resultCount')).toBe(
        10,
      );
      expect(wrapper.find(SearchResultsShownHandler).prop('sectionCount')).toBe(
        1,
      );
    });
  });

  it('does not render empty services results', () => {
    const wrapper = shallow(
      <ConfluencePostQuery
        {...baseProps}
        {...createResults()}
        services={baseProps.services}
        fetchMoreFn={fetchMoreFn}
        teamNames={{}}
      />,
    );

    const sections = wrapper.find(SearchResultSection);

    expect(sections).toHaveLength(2);
    assertSearchResultSectionsHaveTitles(sections, [
      messages.compass_libraries_applications_more_section_heading,
      messages.compass_teams_and_people_section_heading,
    ]);
  });

  it('does not render empty peopleTeams results', () => {
    const wrapper = shallow(
      <ConfluencePostQuery
        {...baseProps}
        {...createResults()}
        peopleTeams={baseProps.peopleTeams}
        fetchMoreFn={fetchMoreFn}
        teamNames={{}}
      />,
    );

    const sections = wrapper.find(SearchResultSection);

    expect(sections).toHaveLength(2);
    assertSearchResultSectionsHaveTitles(sections, [
      messages.compass_services_section_heading,
      messages.compass_libraries_applications_more_section_heading,
    ]);
  });

  it('renders no results screen if no results at all', () => {
    const wrapper = shallow(
      <ConfluencePostQuery
        {...baseProps}
        teamNames={{}}
        fetchMoreFn={fetchMoreFn}
      />,
    );

    const sections = wrapper.find(SearchResultSection);

    expect(sections).toHaveLength(0);
  });

  it('renders a show more link when there are more than 30 service results', () => {
    const wrapper = shallow(
      <ConfluencePostQuery
        {...baseProps}
        services={{
          isLoading: false,
          cursor: 'a cursor',
          results: createServiceResults(31, CompassComponentType.SERVICE),
        }}
        teamNames={{}}
        fetchMoreFn={fetchMoreFn}
      />,
    );
    const sectionWrapper = wrapper
      .find(SearchResultSection)
      .filterWhere(
        (resultSection) =>
          getSearchResultSectionTitleText(resultSection) ===
          messages.compass_services_section_heading.defaultMessage,
      );

    expect(sectionWrapper.find(SearchResultSectionLink).exists()).toBeTruthy();
  });

  it('renders the right show more link when there are more than 13 library etc results', () => {
    const wrapper = shallow(
      <ConfluencePostQuery
        {...baseProps}
        libsAppsMore={{
          isLoading: false,
          cursor: 'a cursor',
          results: createServiceResults(14, CompassComponentType.OTHER),
        }}
        teamNames={{}}
        fetchMoreFn={fetchMoreFn}
      />,
    );
    const sectionWrapper = wrapper
      .find(SearchResultSection)
      .filterWhere(
        (resultSection) =>
          getSearchResultSectionTitleText(resultSection) ===
          messages.compass_libraries_applications_more_section_heading
            .defaultMessage,
      );

    expect(sectionWrapper.find(SearchResultSectionLink).exists()).toBeTruthy();
  });

  it('does not render show more link when all results are loaded', () => {
    const wrapper = shallow(
      <ConfluencePostQuery
        {...baseProps}
        services={{
          isLoading: false,
          cursor: null,
          results: createServiceResults(35, CompassComponentType.SERVICE),
        }}
        teamNames={{}}
        fetchMoreFn={fetchMoreFn}
      />,
    );
    const sectionWrapper = wrapper
      .find(SearchResultSection)
      .filterWhere(
        (w) =>
          getSearchResultSectionTitleText(w) ===
          messages.compass_services_section_heading.defaultMessage,
      );

    expect(sectionWrapper.find(SearchResultSectionLink).exists()).toBeFalsy();
  });
});
