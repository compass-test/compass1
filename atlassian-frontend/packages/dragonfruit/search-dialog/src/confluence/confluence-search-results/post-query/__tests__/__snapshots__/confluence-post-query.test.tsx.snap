// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfluencePostQuery /> render with results matches snapshot - with team names 1`] = `
<span
  data-test-id="search-dialog-compass-post-query"
>
  <div
    className="search-result-section"
    data-test-id="search-result-section-services"
  >
    <SearchResultSection
      title={
        <FormattedMessage
          defaultMessage="Services"
          description="Section heading for the search flyout Services section."
          id="compass_search_dialog.compass.services_section_heading"
          values={Object {}}
        />
      }
    >
      <ProductSearchResult
        analyticContext={
          Object {
            "containerId": "none",
            "globalIndex": 0,
            "id": "2a430a73-d9e9-4bcc-a46a-51618965ffea",
            "indexWithinSection": 0,
            "isCached": false,
            "sectionId": "compass-service",
            "sectionIndex": 0,
          }
        }
        containerDetail="Optional secondary data-warehouse"
        href="https://compass.com/2a430a73-d9e9-4bcc-a46a-51618965ffea?search_id=abc"
        icon={
          <ComponentTypeIcon
            label="Parker - Ward"
            type="SERVICE"
          />
        }
        isCollapsed={false}
        key="item_https://compass.com/2a430a73-d9e9-4bcc-a46a-51618965ffea"
        owner="another team"
        screen="postQuerySearchResults"
        title="Parker - Ward"
      />
    </SearchResultSection>
  </div>
  <div
    className="search-result-section"
    data-test-id="search-result-section-appsnlibs"
  >
    <SearchResultSection
      title={
        <FormattedMessage
          defaultMessage="Libraries, Applications & More"
          description="Section heading text for the search flyout Libraries, Applications & More section."
          id="compass_search_dialog.compass.libraries_applications_more_section_heading"
          values={Object {}}
        />
      }
    >
      <ProductSearchResult
        analyticContext={
          Object {
            "containerId": "none",
            "globalIndex": 1,
            "id": "81098dee-c818-44c9-8bf7-8c1398f8e79c",
            "indexWithinSection": 0,
            "isCached": false,
            "sectionId": "compass-application",
            "sectionIndex": 1,
          }
        }
        containerDetail="Front-line executive algorithm"
        href="https://compass.com/81098dee-c818-44c9-8bf7-8c1398f8e79c?search_id=abc"
        icon={
          <ComponentTypeIcon
            label="Wintheiser LLC"
            type="APPLICATION"
          />
        }
        isCollapsed={false}
        key="item_https://compass.com/81098dee-c818-44c9-8bf7-8c1398f8e79c"
        owner="yet another team"
        screen="postQuerySearchResults"
        title="Wintheiser LLC"
      />
    </SearchResultSection>
  </div>
  <div
    className="search-result-section"
    data-test-id="search-result-section-teams"
  >
    <SearchResultSection
      title={
        <FormattedMessage
          defaultMessage="Teams & People"
          description="Section heading text for search flyout Teams & People section."
          id="compass_search_dialog.compass.teams_and_people_heading"
          values={Object {}}
        />
      }
    >
      <ProductSearchResult
        analyticContext={
          Object {
            "containerId": null,
            "globalIndex": 1,
            "id": "a805ab01-0e05-4cef-8042-cadcad480661",
            "indexWithinSection": 0,
            "isCached": false,
            "sectionId": "compass-team",
            "sectionIndex": 1,
          }
        }
        href="/compass/people/team/a805ab01-0e05-4cef-8042-cadcad480661?search_id=abc"
        icon={
          <ResultAvatar
            appearance="circle"
            borderColor="transparent"
            size="xsmall"
            src="ashlynn.net"
          />
        }
        isCollapsed={false}
        key="people_a805ab01-0e05-4cef-8042-cadcad480661"
        screen="postQuerySearchResults"
        title="Sleek Fresh Chair"
      />
    </SearchResultSection>
  </div>
  <WithAnalyticsEvents(WrappedComponent)
    isLoading={false}
    isPreQuery={false}
    resultCount={1}
    results={Array []}
    sectionCount={1}
  />
</span>
`;

exports[`<ConfluencePostQuery /> render with results matches snapshot 1`] = `
<span
  data-test-id="search-dialog-compass-post-query"
>
  <div
    className="search-result-section"
    data-test-id="search-result-section-services"
  >
    <SearchResultSection
      title={
        <FormattedMessage
          defaultMessage="Services"
          description="Section heading for the search flyout Services section."
          id="compass_search_dialog.compass.services_section_heading"
          values={Object {}}
        />
      }
    >
      <ProductSearchResult
        analyticContext={
          Object {
            "containerId": "none",
            "globalIndex": 0,
            "id": "ed82d4e5-9edf-4ec4-86f4-f733a072edb7",
            "indexWithinSection": 0,
            "isCached": false,
            "sectionId": "compass-service",
            "sectionIndex": 0,
          }
        }
        containerDetail="Quality-focused next generation knowledge user"
        href="https://compass.com/ed82d4e5-9edf-4ec4-86f4-f733a072edb7?search_id=abc"
        icon={
          <ComponentTypeIcon
            label="Klein Inc"
            type="SERVICE"
          />
        }
        isCollapsed={false}
        key="item_https://compass.com/ed82d4e5-9edf-4ec4-86f4-f733a072edb7"
        screen="postQuerySearchResults"
        title="Klein Inc"
      />
    </SearchResultSection>
  </div>
  <div
    className="search-result-section"
    data-test-id="search-result-section-appsnlibs"
  >
    <SearchResultSection
      title={
        <FormattedMessage
          defaultMessage="Libraries, Applications & More"
          description="Section heading text for the search flyout Libraries, Applications & More section."
          id="compass_search_dialog.compass.libraries_applications_more_section_heading"
          values={Object {}}
        />
      }
    >
      <ProductSearchResult
        analyticContext={
          Object {
            "containerId": "none",
            "globalIndex": 1,
            "id": "38d97d60-ec9c-4fc6-96ea-1039c01e1357",
            "indexWithinSection": 0,
            "isCached": false,
            "sectionId": "compass-other",
            "sectionIndex": 1,
          }
        }
        containerDetail="Proactive zero defect moratorium"
        href="https://compass.com/38d97d60-ec9c-4fc6-96ea-1039c01e1357?search_id=abc"
        icon={
          <ComponentTypeIcon
            label="Bergstrom - Schoen"
            type="OTHER"
          />
        }
        isCollapsed={false}
        key="item_https://compass.com/38d97d60-ec9c-4fc6-96ea-1039c01e1357"
        screen="postQuerySearchResults"
        title="Bergstrom - Schoen"
      />
    </SearchResultSection>
  </div>
  <div
    className="search-result-section"
    data-test-id="search-result-section-teams"
  >
    <SearchResultSection
      title={
        <FormattedMessage
          defaultMessage="Teams & People"
          description="Section heading text for search flyout Teams & People section."
          id="compass_search_dialog.compass.teams_and_people_heading"
          values={Object {}}
        />
      }
    >
      <ProductSearchResult
        analyticContext={
          Object {
            "containerId": null,
            "globalIndex": 1,
            "id": "8a292807-7faa-428c-9279-28a0316860c0",
            "indexWithinSection": 0,
            "isCached": false,
            "sectionId": "compass-team",
            "sectionIndex": 1,
          }
        }
        href="/compass/people/team/8a292807-7faa-428c-9279-28a0316860c0?search_id=abc"
        icon={
          <ResultAvatar
            appearance="circle"
            borderColor="transparent"
            size="xsmall"
            src="annabell.com"
          />
        }
        isCollapsed={false}
        key="people_8a292807-7faa-428c-9279-28a0316860c0"
        screen="postQuerySearchResults"
        title="Practical Frozen Sausages"
      />
    </SearchResultSection>
  </div>
  <WithAnalyticsEvents(WrappedComponent)
    isLoading={false}
    isPreQuery={false}
    resultCount={1}
    results={Array []}
    sectionCount={1}
  />
</span>
`;
